src/Observable.ts

```diff
@@ -37,6 +37,13 @@ import {ErrorObservable} from './observable/throw';
 import {AjaxCreationMethod} from './observable/dom/ajax';
 import {WebSocketSubject} from './observable/dom/webSocket';
 
+import {CombineLatestSignature} from './operator/combineLatest';
+
+/* These types would be come standard types used through-out the library */
+export type ObservableOrPromise<T> = Observable<T> | Promise<T>;
```

Comment:
Let's separate this. Thinking about PR like https://github.com/ReactiveX/RxJS/pull/1175, I think there might be enough common usages for these types. Those implementation only might need import this definition, not whole `Observable`.